plasmapy/diagnostics/thomson.py

```diff
@@ -250,6 +251,17 @@ def spectral_density_lite(
     # Apply an instrument function if one is provided
     if instr_func_arr is not None:
         Skw = np.convolve(Skw, instr_func_arr, mode="same")
+
+    # add notch(es) to the spectrum if any are provided
+    if notches is not None:
+        if np.shape(notches) == (2,):
+            notches = np.array([notches])
+
+        for notch in notches:
+            x0 = np.argmin(np.abs(wavelengths - notch[0]))
+            x1 = np.argmin(np.abs(wavelengths - notch[1]))
+            Skw[x0:x1] = 0
```

Comment:
Not sure you want to set it to zero. The way I've handled notches is: 

1) Set it to np.nan, then it will be ignored in the fit. This causes issues when also using an insturment function, though. 
2) Cut it out of both the wavelength and Skw arrays, e.g. 

```
wavelength = np.concatenate( (wavelength[:x0], wavelength[x1:] ))
Skw = np.concatenate( (Skw[:x0], Skw[x1:] ))
```

Reply:
I can see how setting it to 0 in a larger range than the true notch would increase the value of \chi^2 (or some other goodness of fit metric) during fitting, but assuming that the fitting is only trying to find minima of \chi^2, shouldn't it still manage to properly fit the outer wings of the EPW spectrum, because the notch region just adds some fixed contribution to \chi^2? The reason I don't want to just cut it out of both arrays is because I eventually want to make the fitting functions do integral normalization, and by cutting it out of both, the entire trapezoidal region over the notch still contributes to the integral, which then becomes sensitive to noise at the edges of the notch. Also, np.trapz doesn't seem to have an elegant way to ignore nan values, which would also complicate the integral normalization in the nan approach,

Reply:
Hm, I guess that makes sense? Let's try it your way, maybe this will work better.